python-threatexchange/threatexchange/signal_type/pdq/pdq_index2.py

```diff
@@ -0,0 +1,185 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+
+"""
+Implementation of SignalTypeIndex abstraction for PDQ by wrapping
+hashing.pdq_faiss_matcher.
+"""
+
+import typing as t
+import faiss
+import numpy as np
+import pickle
+
+
+from threatexchange.signal_type.index import (
+    IndexMatchUntyped,
+    SignalSimilarityInfoWithIntDistance,
+    SignalTypeIndex,
+    T as IndexT,
+)
+from threatexchange.signal_type.pdq.pdq_faiss_matcher import (
+    PDQMultiHashIndex,
+    PDQFlatHashIndex,
+    PDQHashIndex,
+)
+
+DEFAULT_MATCH_DIST = 31
+DIMENSIONALITY = 256
+
+PDQIndexMatch = IndexMatchUntyped[SignalSimilarityInfoWithIntDistance, IndexT]
+
+
+class PDQIndex2(SignalTypeIndex[IndexT]):
+    """
+    Wrapper around the pdq faiss index lib using PDQMultiHashIndex
+    """
+
+    def __init__(
+        self,
+        threshold: int = DEFAULT_MATCH_DIST,
+        index: t.Optional[faiss.Index] = None,
```

Comment:
nit: swap these and turn threshold into a kw arg

```suggestion
        index: t.Optional[faiss.Index] = None,
        *
        threshold: int = DEFAULT_MATCH_DIST,

```